zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/group/GroupStateTest.java

```diff
@@ -85,4 +85,34 @@ void shouldReturnNullIfNameDoesNotExist() {
     // then
     assertThat(key.isPresent()).isFalse();
   }
+
+  @Test
+  void shouldUpdateGroup() {
+    // given
+    final var groupKey = 1L;
+    final var groupName = "group";
+    final var groupRecord = new GroupRecord().setGroupKey(groupKey).setName(groupName);
+    groupState.create(groupKey, groupRecord);
+
+    final var updatedGroupName = "updatedGroup";
+    groupRecord.setName(updatedGroupName);
+
+    // when
+    groupState.update(groupKey, groupRecord);
+
+    // then
+    var group = groupState.get(groupKey);
+    assertThat(group.isPresent()).isTrue();
+    var persistedGroup = group.get();
+    assertThat(persistedGroup.getGroupKey()).isEqualTo(groupKey);
+    assertThat(persistedGroup.getName()).isEqualTo(updatedGroupName);
+
+    final var groupKeyByName = groupState.getGroupKeyByName(updatedGroupName);
+    assertThat(groupKeyByName.isPresent()).isTrue();
+    group = groupState.get(groupKeyByName.get());
+    assertThat(group.isPresent()).isTrue();
+    persistedGroup = group.get();
+    assertThat(persistedGroup.getGroupKey()).isEqualTo(groupKey);
+    assertThat(persistedGroup.getName()).isEqualTo(updatedGroupName);
```

Comment:
❓ Is this necessary (from line 112 to the end) ? I'd just assert that the key returned by `getGroupKeyByName` is equal to `groupKey`

Reply:
I think it should be asserted since the update is performed in several steps, i.e.:
1. Remove old entry from `groupByNameColumnFamily`;
2. Add new entry from `groupByNameColumnFamily`;
3. Update persistedGroup in `groupColumnFamily`.

If a code change ever changes, for example, step [3], a dev should be made aware of it through the CI.

Reply:
Ah okay, I realized what you mean. I'll remove the code.